src/main/scala/verification/schedulers/DPORwHeuristics.scala

```diff
@@ -1095,11 +1157,19 @@ class DPORwHeuristics(depth_bound: Option[Int] = None) extends Scheduler with La
   def test(events: Seq[ExternalEvent],
            violation_fingerprint: ViolationFingerprint,
            _stats: MinimizationStats) : Option[EventTrace] = {
+    currentSubsequence = events
+    lookingFor = violation_fingerprint
     stats = _stats
     Instrumenter().scheduler = this
-    // TODO(cs): explore schedules.
+    run(events, initialTrace=Some(_initialTrace),
+        initialGraph=Some(_initialDegGraph))
+    traceSem.acquire()
```

Comment:
Rather than mixing where traceSem shows up (the current version makes it so that one cannot use DPORwHeuristics without using this function since traceSem is acquired here and released elsewhere, can we perhaps move to something like https://github.com/NetSys/sts2-applications/blob/panda-rbcast/src/main/scala/mainPing.scala#L54